apps/deploy-web/src/context/CustomChainProvider/CustomChainProvider.tsx

```diff
@@ -12,17 +12,31 @@ import { useChain } from "@cosmos-kit/react";
 import { akash, akashSandbox, akashTestnet, assetLists } from "@src/chains";
 import networkStore from "@src/store/networkStore";
 import { customRegistry } from "@src/utils/customRegistry";
+import { useEffect, useState } from "react";
+import { CosmosMetamaskExtensionWallet } from "@cosmos-kit/cosmos-extension-metamask";
 
 type Props = {
   children: React.ReactNode;
 };
 
 export function CustomChainProvider({ children }: Props) {
+  const [metamask, setMetamask] = useState<CosmosMetamaskExtensionWallet[]>([]);
+
+  useEffect(() => {
+    import("@cosmos-kit/cosmos-extension-metamask").then(module => {
```

Comment:
why not importing it synchronously?

Reply:
I think it's because of nextjs and ssr, the package needs the window obj and can't be imported normally. They should fix...

Reply:
![image](https://github.com/user-attachments/assets/9dca68d0-21fe-45e4-9d63-7c486f8cbe23)

Reply:
It doesn't work as per the documentation https://docs.cosmology.zone/cosmos-kit/integrating-wallets/cosmos-metamask-extension

Reply:
> @schnetzlerjoe Hey, is it possible to make it compatible with nextjs? The error seems to be coming from https://github.com/cosmos/snap/blob/32c3b724f3f2254218181c11572bc18d6afdadaf/packages/snapper/package.json

I can see what I can do!

Reply:
@baktun14 I pushed a new version to snapper https://www.npmjs.com/package/@cosmsnap/snapper which dynamically checks if the package is being loaded on the server or client and will try to attempt to use the appwrite package based on that.

This is a new functionality I have not tested e2e so do let me know if it works. You will need to ensure that cosmos kit is using the version `0.2.7` https://www.npmjs.com/package/@cosmsnap/snapper

Let me know how it goes!

Reply:
I'm having issues applying the override. Is it possible to release a new version of `@cosmos-kit/cosmos-extension-metamask`?

Reply:
@baktun14 Let me push and ping them. But what are you getting in the override?

Reply:
You can checkout the branch and test it out, but from what I understand, it only works for first npm install with a package-lock and there's also issues with workspaces.

Reply:
@baktun14 I got it to work deleting package-lock, setting the below in ./apps/deploy-web/package.json, and running npm i.
```json
  "overrides": {
    "@radix-ui/react-dismissable-layer": "^1.0.5",
    "@radix-ui/react-focus-scope": "^1.0.4",
    "@cosmsnap/snapper": "0.2.7",
    "@cosmos-kit/cosmos-extension-metamask": {
      "@cosmsnap/snapper": "0.2.7"
    }
  }
```

Try that and let me know how it goes.

Reply:
@schnetzlerjoe I prefer not to regenerate the package-lock file as it may have unintended effects to the dependencies of console apps.

Reply:
@schnetzlerjoe Any update on the new version of `cosmos-extension-metamask`?